File: unity_app/Assets/Scripts/Core/Enums/EventEnums.cs
namespace EventPlannerSim.Core
{
    /// <summary>
    /// Represents the current status of an event in its lifecycle.
    /// </summary>
    public enum EventStatus
    {
        Inquiry,
        Accepted,
        Planning,
        Executing,
        Completed,
        Cancelled
    }

    /// <summary>
    /// Represents the complexity level of an event type.
    /// </summary>
    public enum EventComplexity
    {
        Low,
        Medium,
        High,
        VeryHigh
    }

    /// <summary>
    /// Represents the current workload status based on active event count.
    /// </summary>
    public enum WorkloadStatus
    {
        Optimal,
        Comfortable,
        Strained,
        Critical
    }

    /// <summary>
    /// Represents the current phase of event planning/execution.
    /// </summary>
    public enum EventPhase
    {
        Booking,
        PrePlanning,
        ActivePlanning,
        FinalPrep,
        ExecutionDay,
        Results
    }

    /// <summary>
    /// Represents the budget categories for event allocation.
    /// </summary>
    public enum BudgetCategory
    {
        Venue,
        Catering,
        Entertainment,
        Decorations,
        Staffing,
        Contingency,
        Photography,
        Florals,
        Rentals,
        AudioVisual,
        Transportation,
        Security
    }
}

File: EventApp/EventsApp/Models/Enums/EventEnums.swift
import Foundation

enum EventStatus: String, CaseIterable, Codable, Hashable {
    case inquiry
    case accepted
    case planning
    case executing
    case completed
    case cancelled
}

enum EventComplexity: String, CaseIterable, Codable, Hashable {
    case low
    case medium
    case high
    case veryHigh
}

enum WorkloadStatus: String, CaseIterable, Codable, Hashable {
    case optimal
    case comfortable
    case strained
    case critical
}

enum EventPhase: String, CaseIterable, Codable, Hashable {
    case booking
    case prePlanning
    case activePlanning
    case finalPrep
    case executionDay
    case results
}

enum BudgetCategory: String, CaseIterable, Codable, Hashable {
    case venue
    case catering
    case entertainment
    case decorations
    case staffing
    case contingency
    case photography
    case florals
    case rentals
    case audioVisual
    case transportation
    case security
}

enum TaskStatus: String, CaseIterable, Codable, Hashable {
    case locked
    case pending
    case inProgress
    case completed
    case failed
}
